src/third-party-trackers.ts

```diff
@@ -1,118 +1,78 @@
-    }
-};
-export const setupThirdpartyTrackersInspector = async (page, eventDataHandler: (event: BlacklightEvent) => void, blockRequests = false) => {
-    if (blockRequests) {
+export const setUpThirdPartyTrackersInspector = async (
+    page: Page,
+    eventDataHandler: (event: TrackingRequestEvent) => void,
+    enableAdBlock = false
+) => {
+    if (enableAdBlock) {
         await page.setRequestInterception(true);
     }
+
     page.on('request', async request => {
-        let blocked = false;
-        Object.entries(blockers).forEach(([listName, blocker]) => {
-            const {
-                match, // `true` if there is a match
-                filter // instance of NetworkFilter which matched
-            } = blocker.match(fromPuppeteerDetails(request));
-            if (match) {
-                const stack = [
+        let isBlocked = false;
+
+        for (const [listName, blocker] of Object.entries(blockers)) {
+            const { match, filter } = blocker.match(fromPuppeteerDetails(request));
+
+            if (!match) {
+                continue;
+            }
+
+            isBlocked = true;
+
+            const query = new URL(request.url()).searchParams;
+            for (const param in query) {
+                try {
+                    query[param] = JSON.parse(query[param]);
+                } catch {}
+            }
+
+            eventDataHandler({
+                data: {
+                    query,
+                    filter: filter.toString(),
+                    listName
+                },
+                stack: [
                     {
-                        fileName: request.frame() ? request.frame().url() : '',
-                        source: `ThirdPartyTracker RequestHandler`
-                    }
-                ];
-                const parsedUrl = url.parse(request.url());
-                let query = null;
-                if (parsedUrl.query) {
-                    query = decodeURLParams(parsedUrl.query);
-                    for (const param in query) {
-                        if (query.hasOwnProperty(param)) {
-                            query[param] = parseJSONSafely(query[param]);
-                        }
+                        fileName: request.frame()?.url() ?? '',
+                        source: 'ThirdPartyTracker RequestHandler'
                     }
-                }
-                eventDataHandler({
-                    data: {
-                        filter: filter.toString(),
-                        listName,
-                        query
-                    },
-                    stack,
-                    type: 'TrackingRequest',
-                    url: request.url()
-                });
-                if (blockRequests) {
-                    request.abort();
-                    blocked = true;
-                }
+                ],
+                type: 'TrackingRequest',
+                url: request.url()
+            });
+
+            break;
```

Comment:
Was this previously making multiple eventDataHandler() calls if multiple filters matched?

Reply:
No, `PuppeteerBlocker.match()` only returns the first match.